hawkular-wildfly-agent/src/main/java/org/hawkular/agent/monitor/extension/OperationSubsystemStart.java

```diff
@@ -37,22 +37,34 @@ public void execute(OperationContext opContext, ModelNode model) throws Operatio
             MonitorService service = (MonitorService) serviceRegistry.getRequiredService(name).getValue();
 
             boolean restart = model.get("restart").asBoolean(false);
-            if (restart) {
-                LOGGER.debug("Asked to restart the Hawkular Monitor service. Will stop it, then restart it now.");
-                service.stopMonitorService();
-            } else {
-                LOGGER.debug("Asked to start the Hawkular Monitor service");
-            }
-            if (!service.isMonitorServiceStarted()) {
-                Thread newThread = new Thread(new Runnable() {
-                    @Override
-                    public void run() {
+            long delay = model.get("delay").asLong(0L);
+
+            Thread newThread = new Thread(new Runnable() {
+                @Override
+                public void run() {
+                    if (delay > 0) {
+                        try {
+                            Thread.sleep(delay);
+                        } catch (Exception e) {
+                            // just continue with start

```

Comment:
Actually, just return, don't throw exception - don't know if that default exception handler will puke in the log file. just silently return here if the sleep is interrupted.